### PR TITLE
Support UTC DateTime Structures

### PR DESCRIPTION
**Pull request by fbiville**
✅ TestKit PR is: https://github.com/neo4j-drivers/testkit/pull/470
✅ Extra TestKit PR: https://github.com/neo4j-drivers/testkit/pull/495

### CODE REVIEW

File: neo4j/internal/bolt/hydrator_test.go [hunk 1]
```diff
@@ -649,12 +649,46 @@ func TestHydrator(ot *testing.T) {
...
+				// UTC Datetime with time zone name
+				packer.StructHeader('i', 3)
+				packer.Int64(datetime.Unix())
+				packer.Int64(datetime.UnixNano() - (datetime.Unix() * int64(time.Second)))
+				packer.String(zoneName)
+			},
+			x: &db.Record{Values: []interface{}{
+				time.Date(1999, 12, 31, 23, 59, 59, 1, timeZone),
+			}},
+		},
```
Comment (bigmontz) [inline neo4j/internal/bolt/hydrator_test.go:686]: You may could test some summer time, just in case.

**Review by bigmontz — APPROVED** [review]
🌞

File: neo4j/internal/bolt/hydrator_test.go (hunk 1 above)
Comment (fbiville) [reply neo4j/internal/bolt/hydrator_test.go:686]: there are specific tests below with `Thu Jun 16 2022 13:00:00 UTC`

File: neo4j/internal/bolt/hydrator.go [hunk 2]
```diff
@@ -709,25 +717,71 @@ func (h *hydrator) point3d(n uint32) interface{} {
...
+	l, err := time.LoadLocation(zone)
+	if err != nil {
+		h.setErr(&db.ProtocolError{
+			MessageType: "dateTimeNamedZone",
+			Field:       "location",
+			Err:         err.Error(),
+		})
+		return nil
+	}
+	return time.Date(utcTime.Year(), utcTime.Month(), utcTime.Day(), utcTime.Hour(), utcTime.Minute(), utcTime.Second(), utcTime.Nanosecond(), l)
```
Comment (AndyHeap-NeoTech) [inline neo4j/internal/bolt/hydrator.go]: Minor layout niggle: Differing format to the return in dateTimeOffset which puts each element on a new line.

**Review by AndyHeap-NeoTech — APPROVED** [review]
It looks ok to me. However I've not spent so much time deep in thought on this, so perhaps worth getting one of the other driver authors to review as well.

File: neo4j/internal/bolt/hydrator.go (hunk 2 above)
Comment (thelonelyvulpes) [inline neo4j/internal/bolt/hydrator.go]:
```suggestion
	return time.Date(
		utcTime.Year(),
		utcTime.Month(),
		utcTime.Day(),
		utcTime.Hour(),
		utcTime.Minute(),
		utcTime.Second(),
		utcTime.Nanosecond(),
		l
	)
```

**Review by bigmontz — APPROVED** [review]
🍰

**Comment by fbiville** [comment]
The only failing test here is flaky. Merging this.

**Pull request merged**